### PR TITLE
Chapter8

### PR DESCRIPTION
**Pull request by NurAnisAqilah**
ready

### CODE REVIEW

**Comment by NurAnisAqilah** [comment]
ready

File: app/assets/stylesheets/sessions.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Sessions controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/sessions.scss]: remove file that never used

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class SessionsController < ApplicationController
+  def new
+  end
```
Comment (saobv-2223) [inline app/controllers/sessions_controller.rb]:
```suggestion
  def new; end
```

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+    if user&.authenticate params[:session][:password]
+    # Log the user in and redirect to the user's show page.
+    log_in user
+    redirect_to user
```
Comment (saobv-2223) [inline app/controllers/sessions_controller.rb]:
```suggestion
      # Log the user in and redirect to the user's show page.
      log_in user
      redirect_to user
```

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      render :new 
+    end
+  end
+
+  def destroy
+    log_out
+    redirect_to_root_path
+  end
+
+end
```
Comment (saobv-2223) [inline app/controllers/sessions_controller.rb]:
```suggestion
  end
end
```

File: app/helpers/sessions_helper.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    end
+
+    def logged_in?
+        current_user.present?
+    end
+
+    def log_out
+        session.delete(:user_id)
+        @current_user = nil
+    end
```
Comment (saobv-2223) [inline app/helpers/sessions_helper.rb]: indenting 2 spaces

File: app/views/sessions/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+<h1>Log in</h1>
+<div class="row">
+<div class="col-md-6 col-md-offset-3">
+<%= form_for :session, url: login_path do |f| %>
+<%= f.label :email %>
+<%= f.email_field :email, class: "form-control" %>
+<%= f.label :password %>
+<%= f.password_field :password, class: "form-control" %>
+<%= f.submit "Log in", class: "btn btn-primary" %>
+<% end %>
```
Comment (saobv-2223) [inline app/views/sessions/new.html.erb]: indenting 2 spaces

File: app/views/sessions/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+<p><%= t "new_user?" %><%= link_to t("sign_up_now!"), signup_path %></p>
+</div>
+</div>
```
Comment (saobv-2223) [inline app/views/sessions/new.html.erb]:
```suggestion
</div>

```

File: config/routes.rb [hunk 8]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  get 'sessions/new'
```
Comment (saobv-2223) [inline config/routes.rb]: using " and move under root

**Comment by NurAnisAqilah** [comment]
ready

**Comment by manhnt-0960** [comment]
merged

**Pull request merged**